Remove utilities that never got used

tautschnig opened the pull request:
Where possible, commits introducing them were reverted. Other utilities came into existence even before moving to SVN/git.

Some of this code may be used outside the code base - pinging @kroening and @mgudemann for EBMC, @thk123, @smowton, @peterschrammel as your commits got reverted.

tautschnig committed:
Revert "Added utility class to convert strings into expressions"

This reverts commit 318502880f3876d153c56d049153f45154c9764d.

Removing a class that was never used;
unit/analyses/ai/ai_simplify_lhs.cpp shows how to do the parsing without
any helper.

tautschnig committed:
Revert "new exception class"

This reverts commit f29ca0cf2f4f2aedacbad9a7a0e7af4706005a78.

tautschnig committed:
Revert "added pipe_stream class"

This reverts commit 38642cd07df2b8bee87d085ed1da830b87ed51df.

peterschrammel commented on src/goto-programs/property_checker.h:
```diff
@@ -1,55 +0,0 @@
-#include "goto_trace.h"
-#include "goto_model.h"
-
-class property_checkert:public messaget
```
This is used by Deltacheck and 2LS.

tautschnig replied:
What is it good for? Would it be possible to contribute documentation? Why does CBMC/JBMC not use it?

tautschnig replied:
Moved to #2212 for separate discussion.

peterschrammel commented on src/goto-analyzer/static_analyzer.cpp:
```diff
@@ -1,238 +0,0 @@
-
-#include <analyses/interval_domain.h>
-
-class static_analyzert:public messaget
```
This class should really be removed. I think, however, that it is being used by @smowton?

smowton replied:
No, we're using `analysis/static_analysis.h`, which is apparently unrelated.

peterschrammel commented on src/util/memory_limit.h:
```diff
@@ -1,18 +0,0 @@
-\*******************************************************************/
-
-#ifndef CPROVER_UTIL_MEMORY_LIMIT_H
-#define CPROVER_UTIL_MEMORY_LIMIT_H
```
This is used by TG. We can move it there if you want to remove it from this repo.

tautschnig replied:
The code isn't portable so it might make sense to move it? I don't have very strong feelings about it, but given that the code can't be tested in this repository I'd prefer to move it to a place where it _is_ being tested.

tautschnig replied:
Moved to #2214.

peterschrammel commented on src/solvers/prop/prop.h:
```diff
@@ -17,11 +17,9 @@ Author: Daniel Kroening, kroening@kroening.com
 #include <util/message.h>
 #include <util/threeval.h>
 
-#include "prop_assignment.h"
```
literal.h needs to be included here

tautschnig replied:
Will fix.

thk123 commented on unit/testing-utils/c_to_expr.cpp:
```diff
@@ -1,35 +0,0 @@
-/*******************************************************************\
```
This is OK to go

thk123 commented on unit/testing-utils/require_expr.cpp:
```diff
@@ -1,108 +0,0 @@
-
-#include "require_expr.h"
-
-#include <testing-utils/catch.hpp>
```
These are used extensively in unit tests for the TG module. I'm happy to migrate these into TG, however for writing unit tests validating `exprt`s, they are very useful for keeping tests succinct. If we want to expand unit tests in CBMC I suggest keeping them about.

tautschnig replied:
Actually I had been looking at a not-up-to-date version of the tree: since  2348d10f7 `require_expr.h` is being used here as well, so we're good.

thk123 commented:
Will create a TG pointer bump to verify the other reverts don't have an effect once you've lmk what you want to do re: `require_expr`

smowton commented:
@tautschnig I can't see any of my commits reverted here, do you know which ones you were thinking of?

mgudemann commented:
@tautschnig do you want to to test this with EBMC? I also didn't see one of my commits being reverted.

tautschnig commented:
@smowton: It looks like `infix.h` was yours. Will do a proper revert in the next update of this PR.
@mgudemann: No, none of your commits were reverted, but I hoped you could commend on `vcd_goto_tracet`  and `get_modules`.

smowton commented:
@tautschnig aha, we do use `has_infix` in the security repo.

tautschnig commented:
> aha, we do use has_infix in the security repo.

Does that actually improve the code in some way (readability or some other sort of clarity)? Since I don't know your code, I need your guidance on this.

smowton commented:
Sure, a little -- it's a shorthand for "string b occurs at offset i in string a", like `has_prefix` and `has_suffix`.

mgudemann commented on src/util/substitute.h:
```diff
@@ -1,20 +0,0 @@
-/*******************************************************************\
```
substitute is used in TG

tautschnig replied:
Is there a reason not to use `std::replace`?

Edit: is TG actually using it for substring replacement?

Edit 2: if this is seriously being used, then at least a more efficient implementation should be built. Repeatedly searching from the beginning isn't optimal.

thk123 replied:
Yup 6 instances in TG that look like can be easily replaced to `std::replace` - will post here once that fix has been applied

tautschnig replied:
Moved to #2214.

mgudemann commented on src/util/find_macros.h:
```diff
@@ -1,27 +0,0 @@
-/*******************************************************************\
```
`find_macros.h` is used in EBMC

tautschnig replied:
Moved to #2213 to receive your separate approval/further discussion.

mgudemann commented on src/util/get_module.h:
```diff
@@ -1,26 +0,0 @@
-/*******************************************************************\
```
`get_module` is used in EBMC

mgudemann commented on src/util/preprocessor.h:
```diff
@@ -1,43 +0,0 @@
-/*******************************************************************\
```
`preprocessor` is used in EBMC

mgudemann commented:
@tautschnig I searched in EBMC for occurrences of files deleted in your PR and found the above. I haven't yet put in the time / work to get EBMC to compile with current CBMC/develop though

tautschnig commented:
@mgudemann Many thanks for checking. Are there reasons why moving them to EBMC would not be the right thing to do?

mgudemann commented:
@tautschnig not from my side, could be moved to EBMC

@kroening what is your opinion on this?

On 11 April 2018 at 17:58, Michael Tautschnig <notifications@github.com>
wrote:

> @mgudemann <https://github.com/mgudemann> Many thanks for checking. Are
> there reasons why moving them to EBMC would not be the right thing to do?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/diffblue/cbmc/pull/2019#issuecomment-380504743>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ACeHTK8cHhiZmh3CPYsVhkqlaY_Xbekgks5tnigjgaJpZM4TLgrx>
> .
>

tautschnig commented:
I've found some more unused code, but that requires merging #2128 first (hence this won't build for the moment).

tautschnig committed:
Remove unused sorted_vector.h

tautschnig committed:
Remove unused cpp/recursion_counter.h

tautschnig committed:
Remove unused nondet_ifthenelse.{h,cpp}

Partly reverts 055f515bd47004a6f12a898d1c3d2f1486078932.

tautschnig committed:
Remove path_accelerationt interface

This interface only has a single implementation.

tautschnig committed:
Remove unused goto-analyzer/static_analyzer.{h,cpp}

It has been deprecated (and was no longer being built) in
e936c500d09a205aaf26c9f23f8b28bc37487f34.

tautschnig committed:
Remove unused solver/prop/prop_wrapper.h

tautschnig committed:
Remove unused solvers/prop/prop_conv_store.{h,cpp}

tautschnig committed:
Remove prop_assignmentt interface

This interface was just used once.

tautschnig commented:
1. I have moved all those parts that have raised concerns to separate PRs as indicated.
2. I have split up commits so that each of the remaining ones will only remove one module at a time, and should thus be reverted rather easily if a previously unknown use case arises.

mgudemann commented:
@tautschnig I'd say go with removing, we can move the required modules to EBMC

allredj reviewed (commented):
Passed Diffblue compatibility checks (cbmc commit: 471b20f301af02d14167c2de89068e87daffb38b).

peterschrammel reviewed (approved):
No objections
